restricted_site_access.php

```diff
@@ -122,6 +122,32 @@ public static function add_actions() {
 
 		add_filter( 'pre_option_blog_public', array( __CLASS__, 'pre_option_blog_public' ), 10, 1 );
 		add_filter( 'pre_site_option_blog_public', array( __CLASS__, 'pre_option_blog_public' ), 10, 1 );
+		add_filter( 'application_password_is_api_request', array( __CLASS__, 'is_api_request' ) );
+	}
+
+	/**
+	 * Add filter to override application_password_is_api_request filter. This has to be done since rest API isn't been loaded until we call restrict_access.
+	 *
+	 * @param bool $original_value Original value passed by filter.
+	 * @return bool
+	 */
+	public static function is_api_request( $original_value ) {
```

Comment:
Let's defer to WP as soon as possible.

did action also returns a true value while doing the action, naming things is hard :D 

```suggestion
	public static function is_api_request( $original_value ) {
		if ( did_action( 'init' ) ) {
			return $original_value;
		}
```